opennms-webapp/src/test/java/org/opennms/web/tags/DateTimeTagTest.java

```diff
@@ -0,0 +1,115 @@
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.tags;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+
+import javax.servlet.jsp.JspContext;
+import javax.servlet.jsp.JspWriter;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.springframework.mock.web.MockJspWriter;
+
+public class DateTimeTagTest {
+
+    @Test
+    public void shouldOutputeDateTimeIncludingTimeZone() throws IOException {
+        Instant now = Instant.now();
+        test("yyyy-MM-dd'T'kk:mm:ssxxx");
+    }
+
+    @Test
+    public void shouldHonorSystemSettings() throws IOException {
+        String format = "yyy-MM-dd";
+        System.setProperty(DateTimeTag.SYSTEM_PROPERTY_DATE_FORMAT, format);
+        test(format);
+        System.clearProperty(DateTimeTag.SYSTEM_PROPERTY_DATE_FORMAT);
+    }
+
+    public void test(String expectedPattern) throws IOException {
+        Instant now = Instant.now();
+        String output = DateTimeTagInvoker
+                .create()
+                .setInstant(now)
+                .invokeAndGet();
+        DateTimeFormatter formatter = DateTimeFormatter
+                .ofPattern(expectedPattern)
+                .withZone(ZoneId.systemDefault());
+        assertEquals(formatter.format(now), output);
+    }
+
+    // Helper class to be able to test easier
+    private static class DateTimeTagInvoker {
+
+        private DateTimeTag tag;
+        private StringWriter writer;
+
+        private DateTimeTagInvoker(){
+            writer = new StringWriter();
+            JspWriter jspWriter = new MockJspWriter(writer);
+            JspContext jspContext = Mockito.mock(JspContext.class);
+            when(jspContext.getOut()).thenReturn(jspWriter);
+            tag = new DateTimeTag(){
+                @Override
+                protected JspContext getJspContext() {
+                    return jspContext;
+                }
+            };
+        }
+
+        static DateTimeTagInvoker create(){
+            return new DateTimeTagInvoker();
```

Comment:
Can't we just invoke `new DateTimeTagInvoker()` instead of having a `create()` method?

Reply:
done.